server/src/main/java/org/apache/iotdb/db/mpp/execution/config/SetStorageGroupTask.java

```diff
@@ -45,31 +48,39 @@ public SetStorageGroupTask(SetStorageGroupStatement setStorageGroupStatement) {
   @Override
   public ListenableFuture<ConfigTaskResult> execute() {
     SettableFuture<ConfigTaskResult> future = SettableFuture.create();
-    // Construct request using statement
-    TStorageGroupSchema storageGroupSchema = new TStorageGroupSchema();
-    storageGroupSchema.setName(setStorageGroupStatement.getStorageGroupPath().getFullPath());
-    TSetStorageGroupReq req = new TSetStorageGroupReq(storageGroupSchema);
-
-    ConfigNodeClient configNodeClient = null;
-    try {
-      configNodeClient = new ConfigNodeClient();
-      // Send request to some API server
-      TSStatus tsStatus = configNodeClient.setStorageGroup(req);
-      // Get response or throw exception
-      if (TSStatusCode.SUCCESS_STATUS.getStatusCode() != tsStatus.getCode()) {
-        LOGGER.error(
-            "Failed to execute set storage group {} in config node, status is {}.",
-            setStorageGroupStatement.getStorageGroupPath(),
-            tsStatus);
-        future.setException(new StatementExecutionException(tsStatus));
-      } else {
-        future.set(new ConfigTaskResult(TSStatusCode.SUCCESS_STATUS));
+    if (IoTDB.getInstance().isMppClusterMode()) {
+      // Construct request using statement
```

Comment:
Is this ```if-else``` a temporary implementation? I think this judgement should be integrated in a high level framework?

Reply:
Yes, it is a temporary implementation. We need to fix it in the future. Let me add a TODO mark.